### DISCUSSION TITLE
Collapsible toolbar on SQL Lab

### DISCUSSION THREAD

**Discussion by villebro** [apache/superset#24018] [Ideas]
Currently SQL Lab supports collapsing the toolbar (the one with Database/schema/table dropdowns + table schemas) by clicking on the three dots next to the tab label:

![image](https://github.com/apache/superset/assets/33317356/ab370f80-debf-4206-bbcd-7853692b5c90)

The collapsed/expanded state is persisted in the tab state. While this can be convenient if the user only wishes to collapse the toolbar for a subset of tabs, this behavior is in contrast to other areas of the app, where collapsing is usually
1. not persisted
2. happens via a global collapse/expand icon/button:

Expanded state:
![image](https://github.com/apache/superset/assets/33317356/bc93963b-2e0b-497c-8b23-ec1c5ef1d1ae)

Collapsed state:
![image](https://github.com/apache/superset/assets/33317356/bc363f55-9af6-4c0f-aaa1-8af2a640e483)

For the sake of discoverability and uniformity, I propose changing the behavior, so that

1. we use the same collapse/expand components as in Explore/Dashboard
2. we don't persist the state, i.e. SQL Lab always opens up with the toolbar in expanded state
3. remove the tab-specific collapsibility/expandability

WDYT everyone?

**Comment by eschutho** [comment]
cc @yousoph @kasiazjc

**Comment by villebro** [comment]
cc @michael-s-molina @john-bodley @betodealmeida

**Reply by michael-s-molina** [reply]
> 1. we use the same collapse/expand components as in Explore/Dashboard

Agreed. I'm always in favor of making things behave and look consistently throughout the application.

> 2. we don't persist the state, i.e. SQL Lab always opens up with the toolbar in expanded state

I can see a use case where a user does not want to view the Metrics/Columns in Explore or the Database/Tables in SQL Lab. If there's no underlying problem, I would persist the user's choice for all collapse panels (Filters, Explore, SQL Lab, etc). We did something similar when editing a dashboard for the **Show only my charts** option.

> 3. remove the tab-specific collapsibility/expandability

Yes, as long as the layout is adjusted so that the left panel is outside of the tabs.

**Comment by villebro** [comment]
> I can see a use case where a user does not want to view the Metrics/Columns in Explore or the Database/Tables in SQL Lab. If there's no underlying problem, I would persist the user's choice for all collapse panels (Filters, Explore, SQL Lab, etc). We did something similar when editing a dashboard for the Show only my charts option.

I think persisting the collapsed state in SQL Lab can be confusing, as it will be unclear to the user which database and schema they're operating on. I believe the normal user flow looks something like this:
1. User opens SQL Lab and checks which database + schema is selected
2. The user collapses the toolbar to maximize the IDE screen real estate

If the collapsed state is persisted, then the user would likely have to expand and then re-collapse the toolbar before getting to work to verify that they've selected the right db/schema.

**Reply by michael-s-molina** [reply]
This is an interesting point. If every time a user navigates to a tab, the left panel is open, even if I previously closed it, then it makes sense. If however, a user on Tab A collapses the left panel, navigates to Tab B, and navigates back to Tab A and the tab is collapsed, then we'll have the same problem you mentioned. I checked some popular database tools, and they generally offer a left panel that can be collapsed, but at the same time, the database and schema information is always visible, generally on top of the tabs.

**Comment by kerbylane** [comment]
Just my 2 cents as a user: having an icon in the panel would be much nicer than having to dig in a menu.
